File: w1/cash/cash.c
#include <cs50.h>
#include <stdio.h>

int get_cents(int);
int calculate_quarters(int cents);
int calculate_dimes(int cents);
int calculate_nickels(int cents);
int calculate_pennies(int cents);

int main(void)
{
    int cents = get_cents();

    int quarters = calculate_quarters(cents);
    cents %= 25;

    int dimes = calculate_dimes(cents);
    cents %= 10;

    int nickels = calculate_nickels(cents);
    cents %= 5;

    int pennies = calculate_pennies(cents);

    int totalCoins = quarters + dimes + nickels + pennies;

    printf("Quarters: %d\n", quarters);
    printf("Dimes: %d\n", dimes);
    printf("Nickels: %d\n", nickels);
    printf("Pennies: %d\n", pennies);
    printf("Total coins: %d\n", totalCoins);

    return 0;
}

int get_cents(int)
{
    int cents;
    do
    {
        cents = get_int("Enter the number of cents: ");
    }
    while (cents < 0);

    return cents;
}

int calculate_quarters(int cents)
{
    return cents / 25;
}

int calculate_dimes(int cents)
{
    return cents / 10;
}

int calculate_nickels(int cents)
{
    return cents / 5;
}

int calculate_pennies(int cents)
{
    return cents;
}

File: w6/cash.py
from cs50 import get_float

dollars = -1
while dollars <= 0:
    dollars = get_float("Change: ")

cents = round(dollars * 100)

quarters = 0
while cents >= 25:
    quarters = quarters + 1
    cents = cents - 25

dimes = 0
while cents >= 10:
    dimes = dimes + 1
    cents = cents - 10

nickels = 0
while cents >= 5:
    nickels = nickels + 1
    cents = cents - 5

totalCoins = quarters + dimes + nickels + cents

print(totalCoins)
